File: HeeJu/Doit 02-03/app/src/main/java/com/example/lec02_03/MainActivity.java
package com.example.lec02_03;

import androidx.appcompat.app.AppCompatActivity;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    ImageView upImg, downImg;
    Drawable Img;
    static int switching=1;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        upImg = (ImageView)findViewById(R.id.upImg);
        downImg = (ImageView)findViewById(R.id.downImg);

        Resources res = getResources();
        Img = (Drawable)res.getDrawable(R.drawable.bear);

        setImage(null, upImg);


    }

    public void setImage(ImageView bf, ImageView now){
        if (bf != null){
            bf.setImageDrawable(null);
        }

        now.setImageDrawable(Img);
        now.getLayoutParams().width = Img.getIntrinsicWidth();
        now.getLayoutParams().height = Img.getIntrinsicHeight();
    }



    public void onDownBtnClicked(View v){
        if (switching == 1){
            setImage(upImg,downImg);
            switching = 2;
        }
    }

    public void onUpBtnClicked(View v){
        if (switching == 2){
            setImage(downImg,upImg);
            switching = 1;
        }
    }

}

File: Comin/app/src/main/java/org/techtown/comin/MainActivity.kt
package org.techtown.comin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.viewpager.widget.ViewPager
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.ktx.auth
import com.google.firebase.ktx.Firebase
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.bottom.*
import org.techtown.comin.Auth.LoginActivity
import org.techtown.comin.Auth.MyCominActivity
import org.techtown.comin.Zzim.ZzimActivity

class MainActivity : AppCompatActivity() {

    internal lateinit var viewpager: ViewPager
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        auth = FirebaseAuth.getInstance()

        val img = arrayOf(
            R.drawable.ai,
            R.drawable.css,
            R.drawable.html,
            R.drawable.id,
            R.drawable.jpg,
            R.drawable.js,
            R.drawable.mp4,
            R.drawable.pdf,
            R.drawable.php,
            R.drawable.png,
            R.drawable.psd,
            R.drawable.tiff
        )

        val text = arrayOf(
            "ai",
            "css",
            "html",
            "id",
            "jpg",
            "js",
            "mp4",
            "pdf",
            "php",
            "png",
            "psd",
            "tiff"
        )

        val gridviewAdapter = GridviewAdapter(this,img,text)
        gridview.adapter = gridviewAdapter
        gridview.setOnItemClickListener { adapterView, view, i, l ->
            val intent = Intent(this, LectureActivity::class.java)
            startActivity(intent)
        }

        viewpager = findViewById(R.id.viewpager) as ViewPager

        val adapter = ViewPagerAdapter(this)
        viewpager.adapter = adapter

        zzim_icon.setOnClickListener {
            val intent = Intent(this, ZzimActivity::class.java)
            startActivity(intent)
        }

        my_page.setOnClickListener {
            if(auth.currentUser == null){
                val intent = Intent(this, LoginActivity::class.java)
                startActivity(intent)
            }else{
                val intent = Intent(this, MyCominActivity::class.java)
                startActivity(intent)
            }
        }
    }
}